aws/aws_secrets_manager/aws_scm.go

```diff
@@ -66,29 +70,110 @@ func New(
 		Credentials: creds,
 		Region:      &region,
 	}
+
+	return NewFromAWSConfig(config)
+}
+
+func NewFromAWSConfig(config *aws.Config) (*AWSSecretsMgr, error) {
 	sess, err := session.NewSession(config)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create a session: %v", err)
 	}
 	scm := secretsmanager.New(sess)
-	return &awsSecretsMgr{
-		scm:   scm,
-		sess:  sess,
-		creds: creds,
-		asc:   asc,
+	return &AWSSecretsMgr{
+		scm: scm,
 	}, nil
 }
 
-func (a *awsSecretsMgr) String() string {
+func (a *AWSSecretsMgr) String() string {
 	return Name
 }
 
-func (a *awsSecretsMgr) GetSecret(
-	secretId string,
-	keyContext map[string]string,
+func (a *AWSSecretsMgr) Get(_ context.Context, key secrets.SecretKey) (map[string]interface{}, error) {
+	secretID := createSecretId(key)
+	secret, _, err := a.get(secretID)
+	return secret, err
+}
+
+func (a *AWSSecretsMgr) Set(_ context.Context, key secrets.SecretKey, secret map[string]interface{}) error {
+	secretID := createSecretId(key)
+	_, err := a.put(secretID, secret)
+	return err
+}
+
+func (a *AWSSecretsMgr) Delete(_ context.Context, key secrets.SecretKey) error {
```

Comment:
Does your caller not want to specify the retention period for keys?

Reply:
I don't need it now. And I need implement it in a way that is supported by all providers. For example the Vault implementation now does not support retention.